website/static/js/apiApplication.js

```diff
@@ -0,0 +1,271 @@
+/*
+ *  Knockout views for the Developer application pages: List and Create/Edit pages.
+ */
+
+
+'use strict';
+
+// TODO: Some of these may not be actually required. Dependency list ripped from profile view
+var $ = require('jquery');
+var ko = require('knockout');
+var bootbox = require('bootbox');
+require('knockout.validation');
+require('knockout.punches');
+ko.punches.enableAll();
+require('knockout-sortable');
+
+var $osf = require('./osfHelpers');
+var koHelpers = require('./koHelpers');
+require('js/objectCreateShim');
+
+
+/*
+ *  Store the data related to a single API application
+ */
+var ApplicationData = function (data){
+    var self = this;
+    self.clientId = ko.observable();
+    self.clientSecret = ko.observable();
+    self.create_date = ko.observable();
+
+    self.owner = ko.observable();
+    self.name = ko.observable();
+    self.description = ko.observable();
+    self.homeUrl = ko.observable();
+    self.callbackUrl = ko.observable();
+
+    // Unchanging properties
+    self.detailUrl = ko.observable();
+    self.apiUrl = ko.observable();
+
+    // Load in data
+    if (data){
+        self.fromJSON(data)
+    }
+};
+
+// Serialize data for POST request
+ApplicationData.prototype.serialize = function () {
+    var self = this;
+    return {
+        client_id: self.clientId(),
+        client_secret: self.clientSecret(),
+        owner: self.owner(),
+        name: self.name(),
+        description: self.description(),
+        home_url: self.homeUrl(),
+        callback_url: self.callbackUrl()
+    }
+    };
+
+// Load data from JSON
+ApplicationData.prototype.fromJSON = function (data) {
```

Comment:
It might be nice to have this named symmetrically with the serialize function. Either this would be deserialize, or that would be toJSON